src/Type/Definition/NamedTypeTrait.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GraphQL\Type\Definition;
+
+use ReflectionClass;
+
+use function array_key_exists;
+use function preg_replace;
+
+/**
+ * @see NamedType
+ */
+trait NamedTypeTrait
```

Comment:
`use NameType<how to name all these contents?>`. It reads well then.

Reply:
I definitely want to keep `NamedType` in there, as the trait is directly tied to the interface. This pattern approximates multiple inheritance, which is unfortunately not available in PHP. Perhaps `NamedTypeImplementation` will make it more clear?

Reply:
yup 👌🏾